go/ai/generate.go

```diff
@@ -383,7 +383,22 @@ func GenerateWithRequest(ctx context.Context, r api.Registry, opts *GenerateActi
 				return resp, nil
 			}
 
-			return generate(ctx, newReq, currentTurn+1, currentIndex+1)
+			finalResp, err := generate(ctx, newReq, currentTurn+1, currentIndex+1)
+			if err != nil {
+				return nil, err
+			}
+
+			// accumulate Reasoning parts for every tool call
+			if finalResp.Message != nil && resp.Message != nil {
```

Comment:
Can you explain this change?

Reply:
Basically we are explicitly recovering the original reasoning parts from the initial `generate()` call because we were returning a new response without them.

from the previous code:
```go
// newReq had the original reasoning parts
return generate(ctx, newReq, currentTurn+1, currentIndex+1)

// the return value of generate() will be a completely new response message ignoring the initial reasoning process from the model
```

Reply:
I did the fix in this PR but if you want, I can move this to a different PR so it does not become a ninja fix

Reply:
Actually, this behavior might be tightly related to https://github.com/firebase/genkit/issues/3741

Reply:
@apascal07, I decided to remove these fixes from this PR so they get addressed in #3851